### ISSUE TITLE
[Workflows] Use a different hydrate/validate pattern for workflows client and opportunities

### ISSUE THREAD

**Issue by daschi** [Recidiviz/pulse-dashboard#2263]
**What needs to be done? Why does it need to be done?**
A clear and concise description of what task needs to be accomplished and why.

**Additional context**
Add any other context about the task here.

**Comment by macfarlandian** [comment]
stray thought: maybe every opportunity just has a computed `isValid` property that can be used as a filter? 

other stray thought: maybe it's better to put our efforts into upstream validations instead of having the frontend even do this at all. Maybe it just needs to do something very basic like verify that the data exists at all?

**Comment by daschi** [comment]
yeah I'm trying to think what we actually need to validate going forward, and an `isValid` filter sounds like it could prevent the UI from breaking from one unexpected result. More upstream validations like what you added for duplicate clients would be great...maybe we need to list the ways in which there would be failures, like unexpected null values in the results. when I was updating the CR record I did some queries to find out which fields were nullable, but would be good to have that in a validation too.
